SourceryRuntime/Sources/Type.swift

```diff
@@ -53,69 +53,141 @@ import Foundation
+        },
+        isExtension: { $0.definedInType?.isExtension == true },
+        filter: { all, extracted in
+            !all.contains(where: { Self.uniqueMethodFilter($0, rhs: extracted) })
         })
     }
 
+    private static func uniqueMethodFilter(_ lhs: Method, rhs: Method) -> Bool {
+        return lhs.name == rhs.name && lhs.isStatic == rhs.isStatic && lhs.isClass == rhs.isClass
+    }
+
+    // sourcery: skipEquality, skipDescription
     /// Subscripts defined in this type only, inluding subscripts defined in its extensions,
     /// but not including subscripts inherited from superclasses (for classes only) and protocols
-    public var subscripts: [Subscript]
+    public var subscripts: [Subscript] {
+        unique({ $0.rawSubscripts }, filter: Self.uniqueSubscriptFilter)
+    }
+
+    /// Unfiltered (can contain duplications from extensions) Subscripts defined in this type only, inluding subscripts defined in its extensions,
+    /// but not including subscripts inherited from superclasses (for classes only) and protocols
+    public var rawSubscripts: [Subscript]
 
     // sourcery: skipEquality, skipDescription
     /// All subscripts defined for this type, including subscripts defined in extensions,
     /// in superclasses (for classes only) and protocols
     public var allSubscripts: [Subscript] {
-        return flattenAll({ $0.subscripts })
+        return flattenAll({ $0.subscripts },
+            isExtension: { $0.definedInType?.isExtension == true },
+            filter: { all, extracted in
+                !all.contains(where: { Self.uniqueSubscriptFilter($0, rhs: extracted) })
+            })
+    }
+
+    private static func uniqueSubscriptFilter(_ lhs: Subscript, rhs: Subscript) -> Bool {
+        return lhs.parameters == rhs.parameters && lhs.returnTypeName == rhs.returnTypeName && lhs.readAccess == rhs.readAccess && lhs.writeAccess == rhs.writeAccess
     }
 
     // sourcery: skipEquality, skipDescription, skipJSExport
     /// Bytes position of the body of this type in its declaration file if available.
     public var bodyBytesRange: BytesRange?
 
-    private func flattenAll<T>(_ extraction: @escaping (Type) -> [T], filter: (([T], T) -> Bool)? = nil) -> [T] {
+    private func flattenAll<T>(_ extraction: @escaping (Type) -> [T], isExtension: (T) -> Bool, filter: ([T], T) -> Bool) -> [T] {
         let all = NSMutableOrderedSet()
-        all.addObjects(from: extraction(self))
-
-        let filteredExtraction = { (target: Type) -> [T] in
-            if let filter = filter {
-                // swiftlint:disable:next force_cast
-                let all = all.array as! [T]
-                let extracted = extraction(target).filter({ filter(all, $0) })
-                return extracted
+        let allObjects = extraction(self)
+
+        /// The order of importance for properties is:
+        /// Base class
+        /// Inheritance
+        /// Protocol conformance
+        /// Extension
```

Comment:
```suggestion
        // The order of importance for properties is:
        // Base class
        // Inheritance
        // Protocol conformance
        // Extension
```

Reply:
Also maybe specify whether this is from highest to lowest or lowest to highest

Reply:
yeah I did in changelog